tools/pylib/_boutcore_build/boutcore.pyx.in

```diff
@@ -265,8 +264,18 @@ cdef class $ftype:
             ny_=ny-2*nyg
             ystart=ny_*nyi
             kwargs['yind']=[ystart,ystart+ny]
-        data=outputs[name]
         dims=outputs.dimensions[name]
+        if outputs._caching:
+            # if BoutOutputs object is caching the arrays, let it collect all
+            # time points, then slice here
+            if dims[0] == 't':
+                data=outputs[name][tind:tind+1]
```

Comment:
Why `[tind:tind+1]` as opposed to just `[tind]` here?

Reply:
Setting the `tind` argument of `collect` to an int doesn't reduce the array dimension; using `[tind:tind+1]` returns data with the same shape as it would be from the `outputs._caching==False` case. Could move 281-2 into the `else` clause instead?

Reply:
Ah, I see! A comment here to that effect would be fine